### ISSUE TITLE
Pymei might crash inside rodan

### ISSUE THREAD

**Issue by noahbaxter** [DDMAL/JSOMR2MEI#3]
When calling various pymei functions and elements,  errors occur showing unhelpful debug information that doesn't return consistent traceback lines. The version of pymei cannot be determined through the script. This might mean that the version of pymei in `rodan` is not up-to-date or not properly installed.

**Comment by vigliensoni** [comment]
You could check how libmei and pymei are installed and set up in the Docker
container. If you find a bug, make the change and create the pull request.

**Comment by noahbaxter** [comment]
for reference, here is the error I get


```
rodan_1         | [2018-05-31 14:45:53,655: ERROR/MainProcess] Task JSOMR to MEI[079622e0-bf30-414f-950f-c5ba42dd88b3] raised unexpected: TypeError('Python argument types in\n    MeiElement.addAttribute(MeiElement, str, unicode)\ndid not match C++ signature:\n    addAttribute(mei::MeiElement {lvalue}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)\n    addAttribute(mei::MeiElement {lvalue}, mei::MeiAttribute*)',)
rodan_1         | Traceback (most recent call last):
rodan_1         |   File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 240, in trace_task
rodan_1         |     R = retval = fun(*args, **kwargs)
rodan_1         |   File "/usr/local/lib/python2.7/dist-packages/celery/app/trace.py", line 438, in __protected_call__
rodan_1         |     return self.run(*args, **kwargs)
rodan_1         |   File "/code/rodan/rodan/jobs/base.py", line 462, in run
rodan_1         |     retval = self.run_my_task(inputs, settings, arg_outputs)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/JSOMR2MEI.py", line 55, in run_my_task
rodan_1         |     mei_string = mei_obj.run()
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 16, in run
rodan_1         |     return self._conversion()
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 23, in _conversion
rodan_1         |     doc = self._createDoc()
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 31, in _createDoc
rodan_1         |     self._generate_mei(doc)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 59, in _generate_mei
rodan_1         |     self._generate_music(el)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 71, in _generate_music
rodan_1         |     self._generate_body(el)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 90, in _generate_body
rodan_1         |     self._generate_mdiv(el)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 97, in _generate_mdiv
rodan_1         |     self._generate_score(el)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 104, in _generate_score
rodan_1         |     self._generate_section(el)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 120, in _generate_section
rodan_1         |     self._generate_staff(el, i)     # generate multiple staves
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 138, in _generate_staff
rodan_1         |     self._generate_layer(el)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 153, in _generate_layer
rodan_1         |     self._generate_clef(el, g)
rodan_1         |   File "/code/rodan/rodan/jobs/JSOMR2MEI/MeiOutput.py", line 169, in _generate_clef
rodan_1         |     el.addAttribute('shape', glyph['glyph']['name'].split('.')[1].upper())
rodan_1         | Exception: Python argument types in
rodan_1         |     MeiElement.addAttribute(MeiElement, str, unicode)
rodan_1         | did not match C++ signature:
rodan_1         |     addAttribute(mei::MeiElement {lvalue}, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >)
rodan_1         |     addAttribute(mei::MeiElement {lvalue}, mei::MeiAttribute*)
```

**Comment by noahbaxter** [comment]
I've confirmed that rodan uses libmei 2.0, whereas JSOMR2MEI relies upon functionality in 3.0. Therefore, the version of libmei (and by extension pymei) inside Rodan need to be updated.

**Comment by noahbaxter** [comment]
This only happens in Python 2.Turns out that this is because MeiElement (and most pymei functions) are expected as strings and not unicode variables. Simply casting the variables as strings solves the problem. Fixed in 53cdd81bfca2fdf293ef07c8f2536d5af5531370
